source/assoc.js

```diff
@@ -22,7 +22,7 @@ import _curry3 from './internal/_curry3';
  *      R.assoc('c', 3, {a: 1, b: 2}); //=> {a: 1, b: 2, c: 3}
  */
 var assoc = _curry3(function assoc(prop, val, obj) {
-  var result = {};
+  var result = obj === undefined || obj === null ? {} : Object.create(obj.__proto__);
```

Comment:
I think `Object.getPrototypeOf()` is the recommended way to access the prototype, and it has broad browser support.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getPrototypeOf#Browser_compatibility